packages/open-truss/src/components/OTRestDataProvider.tsx

```diff
@@ -0,0 +1,103 @@
+import { mapValues, template } from 'lodash'
+
+import {
+  withChildren,
+  BaseOpenTrussComponentV1PropsShape,
+  type BaseOpenTrussComponentV1,
+} from '../configuration/engine-v1'
+import {
+  ObjectSignal,
+  NumberSignal,
+  StringSignal,
+  UnknownSignal,
+  signalValueShape,
+  useSignalEffect,
+} from '../signals'
+import { z } from 'zod'
+
+interface SynchronousRestResult {
+  status: number
+  headers: Record<string, string[]>
+  body: any
+}
+
+export const Props = BaseOpenTrussComponentV1PropsShape.extend({
+  ...withChildren,
+  source: StringSignal,
+  path: z.string(), // TODO: How to handle cases where value could be a `Signal` OR a `String`?
+  method: StringSignal,
+  headers: ObjectSignal,
+  path_values: z.object({}).default({}),
+  force_query: NumberSignal,
+  output: z.array(UnknownSignal).optional(),
+})
+
+const OTRestDataProvider: BaseOpenTrussComponentV1<z.infer<typeof Props>> = (
+  props,
+) => {
+  const {
+    source,
+    path: templatePath,
+    path_values, // eslint-disable-line @typescript-eslint/naming-convention
```

Comment:
I actually prefer camel case but I went with snake case because that was the precedent set by `force_query`. I'm down to change both to camel case though. What think you?

Reply:
Let's update this to camelCase then and change synchronous-uqi-query to also be camelCase. https://chatgpt.com/share/8298ddf1-c4d5-46be-91cb-57ae6d6bb6d2

Reply:
Done ✅